libbeat/common/docker/watcher_test.go

```diff
@@ -12,15 +12,17 @@ import (
 
 type MockClient struct {
 	// containers to return on ContainerList call
-	containers []types.Container
+	containers [][]types.Container
 	// event list to send on Events call
 	events []interface{}
 
 	done chan interface{}
 }
 
 func (m *MockClient) ContainerList(ctx context.Context, options types.ContainerListOptions) ([]types.Container, error) {
-	return m.containers, nil
+	res := m.containers[0]
```

Comment:
Can you elaborate on this change?

Reply:
I added support for several requests to this mock, each request returns the next response and pops it out the list